sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/SwitchShouldNotBeNested.cs

```diff
@@ -42,6 +42,19 @@ public Program(char choice, int value)
                     Console.WriteLine("Other");
                     break;
             }
+
+            string first = "foo", second = "bar";
+
+            var result = first switch
+            {
+                "a" => second switch // Compliant - FN: should check also switch expressions?
```

Comment:
I'm not sure it should apply to switch expressions... at least MS gives example w/ nested switch expressions on their docs site, like a pattern that can be used. 

We can discuss whether it's a nice pattern or not, but nested switch expressions are less bad then nested switch statements (where you can have break, early return, multiple statements per case). Switch expressions have a single return.

@nicolas-harraudeau-sonarsource - WDYT?

Reply:
Indeed, nested expressions are hard to read but not as bad as the statements. 

I've removed the "FN" comment but I've let the test to document the current behavior. We can decide later if we want to extend the rule to cover also this.